### PR TITLE
[16.0][ADD] pos_financial_risk

### PR DESCRIPTION
**Pull request by geomer198**
Extends Partner Financial Risk to manage POS  orders.
If POS order exceeds partner limit POS order forbidden to be confirmed.

### CODE REVIEW

File: pos_financial_risk/static/src/js/ProductScreen.js
Comment (ivs-cetmix) [inline pos_financial_risk/static/src/js/ProductScreen.js]: Cannot confirm order

File: pos_financial_risk/static/src/js/ProductScreen.js
Comment (ivs-cetmix) [inline pos_financial_risk/static/src/js/ProductScreen.js]: Order total exceeds customer credit limit

**Review by ivs-cetmix — CHANGES_REQUESTED** [review]

File: pos_financial_risk/static/src/js/ProductScreen.js
Comment (GabbasovDinar) [inline pos_financial_risk/static/src/js/ProductScreen.js]: please use the new method to define the js files, this will be more flexible for porting to a new version of odoo in the future

```js
/** @odoo-module **/


import ProductScreen from "point_of_sale.ProductScreen";
import Registries from "point_of_sale.Registries";
import { _t } from "web.core";

export const ProductScreenRisk = (ProductScreen) =>
```
and etc

File: pos_financial_risk/static/src/js/ProductScreen.js
Comment (GabbasovDinar) [inline pos_financial_risk/static/src/js/ProductScreen.js]:
```suggestion
            async _onClickPay() {
                const order = this.env.pos.get_order();
                const partner = order.partner;
                if (partner &&  order.get_total_with_tax() + order.get_rounding_applied() > partner.risk_total) {
                    return await this.showPopup("ErrorPopup", {
                        title: _t("Cannot confirm order"),
                        body: _t("Order total exceeds customer credit limit"),
                    });
                }
                return await super._onClickPay();              
            }
```

**Review by GabbasovDinar — CHANGES_REQUESTED** [review]

**Review by chienandalu — COMMENTED** [review]
Probably a silly question :sweat_smile: : PoS orders are paid in the moment, shouldn't that keep the customer credit always balanced?

**Comment by ivs-cetmix** [comment]
> Probably a silly question 😅 : PoS orders are paid in the moment, shouldn't that keep the customer credit always balanced?

Well that's a good question. Actually they should. However at some moment you would like to block a customer from purchasing anything before he pays his debt. Otherwise he can keep paying through POS without taking care of his pending balance 😅

**Comment by luke-stdev001** [comment]
@geomer198 ,

I think this PR is useful and should be re-opened. We have a use-case for it internally and may want to flesh out the concept a little more. How can I contact you to discuss funding this development?

**Comment by ivs-cetmix** [comment]
> @geomer198 ,
> 
> I think this PR is useful and should be re-opened. We have a use-case for it internally and may want to flesh out the concept a little more. How can I contact you to discuss funding this development?

Hi @luke-stdev001 
We will reopen the PR. Pls ping us on odooapps@cetmix.com

**Comment by luke-stdev001** [comment]
@geomer198 ,

Functional testing has been done and it works as expected, thanks for the changes. From my end we are ok to merge pending a code review, however we'd prefer to just pull in the OCA/POS repo rather than maintain our own version until this PR is merged here. 

Would a code review from our team on this PR here help with the OCA process, or are we waiting on PSC and tests to pass for approval to merge into 16.0 for the OCA?

**Comment by ivs-cetmix** [comment]
> @geomer198 ,
> 
> Functional testing has been done and it works as expected, thanks for the changes. From my end we are ok to merge pending a code review, however we'd prefer to just pull in the OCA/POS repo rather than maintain our own version until this PR is merged here.
> 
> Would a code review from our team on this PR here help with the OCA process, or are we waiting on PSC and tests to pass for approval to merge into 16.0 for the OCA?

@luke-stdev001 you can make a code review meanwhile. We will also do our own too

**Review by GabbasovDinar — APPROVED** [review]
LGTM

**Review by ivs-cetmix — APPROVED** [review]
Functional OK

File: test-requirements.txt
Comment (ivantodorovich) [inline test-requirements.txt]: Could please you rebase and update this dependency? 🙏🏻 it seems it was replaced by another PR

- https://github.com/OCA/credit-control/pull/272
- https://github.com/OCA/credit-control/pull/292

File: pos_financial_risk/static/src/xml/PaymentScreen.xml
Comment (ivantodorovich) [inline pos_financial_risk/static/src/xml/PaymentScreen.xml]: It's not really a good practice to add inline styles.
The convention is to add styles through css or scss files instead

**Review by ivantodorovich — APPROVED** [review]

**Comment by ivantodorovich** [comment]
thanks!

/ocabot merge nobump

**Pull request merged**
